### PR TITLE
Fix to use natural sort for flytekit Sagemaker script

### PR DESCRIPTION
**Pull request by convexquad**
# TL;DR
My test SageMaker jobs were failing, but Chang-Hong's jobs worked. After investigating, we discovered that the issue is a text sort that should be a "natural" order sort.

Details for the issue in https://lyft.slack.com/archives/CUMH80X6W/p1604517799068200 and an explanation of this fix in https://lyft.slack.com/archives/CUMH80X6W/p1604526254078800.

JIRA link: https://jira.lyft.net/browse/AVSW-64440

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Without this fix the error can be reproduced in a container with `flytekit` 0.14.x with the following:
```
root@4a9ab1c1ada7:/usr/local/bin# /usr/local/bin/python /usr/local/bin/flytekit_sagemaker_runner.py --__FLYTE_CMD_0_pyflyte-execute__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_10_s3://lyft-modelbuilder/88/q8i35ntlmg-jadoo-task-0__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_1_--task-module__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_2_jadoo.flyte__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_3_--task-name__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_4_jadoo_training_task__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_5_--inputs__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_6_s3://lyft-modelbuilder/metadata/propeller/production/avmlworkflows-development-q8i35ntlmg/jadoo-task/data/inputs.pb__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_7_--output-prefix__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_8_s3://lyft-modelbuilder/metadata/propeller/production/avmlworkflows-development-q8i35ntlmg/jadoo-task/data/0__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_9_--raw-output-data-prefix__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_ENV_VAR_FLYTE_INTERNAL_CONFIGURATION_PATH__ /opt/avjadoo/flytekit.config --__FLYTE_ENV_VAR_FLYTE_INTERNAL_DOMAIN__ development --__FLYTE_ENV_VAR_FLYTE_INTERNAL_IMAGE__ 173840052742.dkr.ecr.us-east-1.amazonaws.com/avmlworkflows:abain_jadoo_1604522520 --__FLYTE_ENV_VAR_FLYTE_INTERNAL_PROJECT__ avmlworkflows --__FLYTE_ENV_VAR_FLYTE_INTERNAL_VERSION__ abain_jadoo_1604522520 --__FLYTE_ENV_VAR_FLYTE_STATSD_DISABLED__ True

Error: --raw-output-data-prefix option requires an argument
Traceback (most recent call last):
  File "/usr/local/bin/flytekit_sagemaker_runner.py", line 94, in <module>
    run(sys.argv)
  File "/usr/local/bin/flytekit_sagemaker_runner.py", line 90, in run
    subprocess.run(flyte_cmd, stdout=sys.stdout, stderr=sys.stderr, encoding="utf-8", check=True)
  File "/usr/local/lib/python3.8/subprocess.py", line 512, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['pyflyte-execute', '--task-module', 's3://lyft-modelbuilder/88/q8i35ntlmg-jadoo-task-0', 'jadoo.flyte', '--task-name', 'jadoo_training_task', '--inputs', 's3://lyft-modelbuilder/metadata/propeller/production/avmlworkflows-development-q8i35ntlmg/jadoo-task/data/inputs.pb', '--output-prefix', 's3://lyft-modelbuilder/metadata/propeller/production/avmlworkflows-development-q8i35ntlmg/jadoo-task/data/0', '--raw-output-data-prefix']' returned non-zero exit status 2.
```
I made this fix to my local container and verified that now the command executed with `subprocess` is formatted correctly:
```
root@4a9ab1c1ada7:/usr/local/bin# /usr/local/bin/python /usr/local/bin/flytekit_sagemaker_runner.py --__FLYTE_CMD_0_pyflyte-execute__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_10_s3://lyft-modelbuilder/88/q8i35ntlmg-jadoo-task-0__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_1_--task-module__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_2_jadoo.flyte__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_3_--task-name__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_4_jadoo_training_task__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_5_--inputs__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_6_s3://lyft-modelbuilder/metadata/propeller/production/avmlworkflows-development-q8i35ntlmg/jadoo-task/data/inputs.pb__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_7_--output-prefix__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_8_s3://lyft-modelbuilder/metadata/propeller/production/avmlworkflows-development-q8i35ntlmg/jadoo-task/data/0__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_CMD_9_--raw-output-data-prefix__ __FLYTE_CMD_DUMMY_VALUE__ --__FLYTE_ENV_VAR_FLYTE_INTERNAL_CONFIGURATION_PATH__ /opt/avjadoo/flytekit.config --__FLYTE_ENV_VAR_FLYTE_INTERNAL_DOMAIN__ development --__FLYTE_ENV_VAR_FLYTE_INTERNAL_IMAGE__ 173840052742.dkr.ecr.us-east-1.amazonaws.com/avmlworkflows:abain_jadoo_1604522520 --__FLYTE_ENV_VAR_FLYTE_INTERNAL_PROJECT__ avmlworkflows --__FLYTE_ENV_VAR_FLYTE_INTERNAL_VERSION__ abain_jadoo_1604522520 --__FLYTE_ENV_VAR_FLYTE_STATSD_DISABLED__ True

Launching command: ['pyflyte-execute', '--task-module', 'jadoo.flyte', '--task-name', 'jadoo_training_task', '--inputs', 's3://lyft-modelbuilder/metadata/propeller/production/avmlworkflows-development-q8i35ntlmg/jadoo-task/data/inputs.pb', '--output-prefix', 's3://lyft-modelbuilder/metadata/propeller/production/avmlworkflows-development-q8i35ntlmg/jadoo-task/data/0', '--raw-output-data-prefix', 's3://lyft-modelbuilder/88/q8i35ntlmg-jadoo-task-0']
```
You can see that in the fixed version `'--raw-output-data-prefix', 's3://lyft-modelbuilder/88/q8i35ntlmg-jadoo-task-0'` displays at the end. In the broken version, this key and its value do not appear together in the command to be executed.

This change means that the code includes a direct dependency on `natsort`. We should update the `requirements.in` file AND include a dependency on `natsort` in `setup.py`. I'm now sure how these are done in `flytekit`, I may have to leave that up to the team.

### CODE REVIEW

**Review by kumare3 — DISMISSED** [review]
LGTM, thank you! Lets wait for all green

**Comment by wild-endeavor** [comment]
bump version in flytekit/__init__.py

**Comment by convexquad** [comment]
Relevant JIRA: https://jira.lyft.net/browse/AVSW-64440

**Review by wild-endeavor — APPROVED** [review]
